src/pyhf/contrib/viz/brazil.py

```diff
@@ -27,24 +27,124 @@ def plot_results(ax, mutests, tests, test_size=0.05):
+    ax.set_xlabel(x_label)
+    ax.set_ylabel(y_label)
+
+
+def plot_cls_components(
+    ax, mutests, tests, test_size=0.05, clb_only=False, clsb_only=False, **kwargs
+):
+    r"""
+    Plot the values of :math:`\mathrm{CL}_{s+b}` and :math:`\mathrm{CL}_{b}`
+    --- the components of the :math:`\mathrm{CL}_{s}` ratio --- on top of the
+    :math:`\mathrm{CL}_{s}` values for a series of hypothesis tests for various
+    POI values.
+
+    Example:
+
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+        >>> import pyhf
+        >>> import pyhf.contrib.viz.brazil
+        >>> pyhf.set_backend("numpy")
+        >>> model = pyhf.simplemodels.hepdata_like(
+        ...     signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0]
+        ... )
+        >>> observations = [51, 48]
+        >>> data = observations + model.config.auxdata
+        >>> poi_vals = np.linspace(0, 5, 41)
+        >>> results = [
+        ...     pyhf.infer.hypotest(
+        ...         test_poi, data, model, return_expected_set=True, return_tail_probs=True
+        ...     )
+        ...     for test_poi in poi_vals
+        ... ]
+        >>> fig, ax = plt.subplots()
+        >>> pyhf.contrib.viz.brazil.plot_cls_components(ax, poi_vals, results)
+
+    Args:
+        ax (:obj:`matplotlib.axes.Axes`): The matplotlib axis object to plot on.
+        mutests (:obj:`list` or :obj:`array`): The values of the POI where the
+          hypothesis tests were performed.
+        tests (:obj:`list` or :obj:`array`): The collection of :math:`p`-values
+          from the hypothesis tests.
+          ``tests`` is required to have the same structure as
+          :func:`pyhf.infer.hypotest`'s return when using ``return_expected_set=True``
+          and ``return_tail_probs=True``: a tuple of :math:`\mathrm{CL}_{s}`,
+          :math:`\left[\mathrm{CL}_{s+b}, \mathrm{CL}_{b}\right]`,
+          :math:`\mathrm{CL}_{s,\mathrm{exp}}` band.
+        test_size (:obj:`float`): The size, :math:`\alpha`, of the test.
+        clb_only (:obj:`bool`): Bool for plotting only the :math:`\mathrm{CL}_{b}`
+          component.
+        clsb_only (:obj:`bool`): Bool for plotting only the :math:`\mathrm{CL}_{s+b}`
+          component.
+    """
+
+    if len(tests[0]) != 3:
+        raise ValueError(
+            f"The components of 'tests' should have len of 3 but have len {len(tests[0])}."
+            + "\n'tests' should have format of: CLs_obs, [CLsb, CLb], [CLs_exp band]"
+        )
+
+    # split into components
+    CLs_obs = np.array([test[0] for test in tests])
+    tail_probs = np.array([test[1] for test in tests])
+    CLs_exp_set = np.array([test[2] for test in tests])
+
+    # zip CLs_obs and CLs_exp_set back into format for plot_results
+    CLs_results = [(obs, exp_set) for obs, exp_set in zip(CLs_obs, CLs_exp_set)]
+
+    # plot CLs_obs and CLs_expected set
+    y_label = kwargs.pop("ylabel", r"$p\,$-value")
```

Comment:
Very nit-picky, but maybe some phrasing to acknowledge that CLs is not a p-value is useful. `(modified) p-value` maybe?

Reply:
Hm. I think the only concern that I have here is that the `CL_s+b` and the `CL_b` _are_ `p-values` and it is just the `CL_s` that is modified as it the ratio. This is a problem with trying to plot both the components _and_ the CLs values all at once &mdash; they're different things.

Reply:
This is part of the reason that @lukasheinrich eventually suggested that we go with [`return_tail_probs` for `hypotest`](https://pyhf.readthedocs.io/en/v0.6.1/_generated/pyhf.infer.hypotest.html) as we had been going back and forth for days about how to describe the returned objects. I'm not sure how to name things better here though.